### ISSUE TITLE
Integrate loss functions with autodiff using custom_gradient

### ISSUE THREAD

**Issue by sahilpatelsp** [google/qhbm-library#51]
It would be a good idea to wrap the QMHL and VQT loss functions with @tf.custom_gradient so that we can integrate these loss functions into standard TensorFlow autodiff.
